varinfo/variable.py

```diff
@@ -423,41 +350,22 @@ def _get_data_type(self, variable: ET.Element) -> str:
         """Extract a string representation of the variable data type."""
         return variable.tag.lstrip(self.namespace).lower()
 
-    def _get_shape(self, variable: ET.Element) -> Tuple[int]:
+    def _get_shape(self, variable: ET.Element) -> tuple[int]:
         """Extract the shape of the variable data array. This is not yet
         implemented as the Dimension information is currently unavailable
         to the Variable XML content.
 
         """
 
-    def _get_attributes(self, variable: ET.Element) -> Dict:
```

Comment:
These abstract methods are now implemented in the `AttributeContainerFromDmr` class. (Similar story below for `VariableFromNetCDF4`)